### ISSUE TITLE
LFS can track and clean .gitattributes in subfolder, breaking itself

### ISSUE THREAD

**Issue by HannesFranke-smartoptics** [git-lfs/git-lfs#3894]
This is related to #1099 but happens when a whole folder is tracked by lfs and a .gitattributes file is added in it.

Basically while lfs prevents you from tracking a .gitattributes file directly it doesn't prevent you from tracking folder which contains a .gitattributes file.

```
mkdir test_repo && cd test_repo
git init
git lfs install

echo 'subfolder/** filter=lfs diff=lfs merge=lfs' > .gitattributes
git add .gitattributes
git commit -m "top .gitattributes"

mkdir subfolder
echo '# test' > subfolder/.gitattributes
git add .
git commit -m "add subfolder"
```

Now `subfolder/.gitattributes` is tracked by lfs leading to exactly the same problems as in #1099 

```
git checkout HEAD^
git checkout master
https://git-lfs.github.com/spec/v1 is not a valid attribute name: subfolder/.gitattributes:1
sha256:e621bcce928241db8b0609ac56504cd1e81a1d7d9625d6c00eca21d2de76295d is not a valid attribute name: subfolder/.gitattributes:2
Previous HEAD position was 18ae692 top .gitattributes
Switched to branch 'master'
```

**Comment by bk2204** [comment]
Hey,

While we can restrict what patterns are given to `git lfs track`, we can't restrict what patterns are given in the `.gitattributes` file.  When you do that, you're basically telling Git that all of those files are to be filtered using Git LFS, and Git will ask for those files to be filtered itself without necessarily telling us what the filename is, so there may or may not be a way we can refuse to filter that file.

In other words, this is a limitation of how Git works.  In general, it's not useful to filter `.gitattributes` files, but there are of course some cases where it could be (e.g., to remove trailing whitespace or sort), so Git doesn't forbid it altogether.

We _can_ refuse to process a pattern that matches a `.gitattributes` file when passed to `git lfs track`, and I can mark this as an enhancement to implement that.

**Comment by magneticflux-** [comment]
I think I encountered this issue when making a `.gitattributes` file like this:

```gitattributes
* filter=lfs diff=lfs merge=lfs -text
```

I wanted to place one of these files in each directory I wanted tracked by Git LFS. Committing, importing, and pushing worked, but when I tried to clone later it didn't know what files to checkout via Git LFS and just left the text placeholder files instead.
